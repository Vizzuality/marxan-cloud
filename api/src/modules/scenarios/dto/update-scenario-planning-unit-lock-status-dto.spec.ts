import { validateSync } from 'class-validator';
import {
  getDtoWithInvalidUuids,
  getDtoWithInvalidMultiPolygon,
} from './__mocks__/dtos.data';

describe(`when changing by id`, () => {
  describe(`when providing invalid uuid`, () => {
    it(`should return error`, () => {
      const result = validateSync(getDtoWithInvalidUuids())[0].children[0];
      expect(result.constraints).toMatchInlineSnapshot(`
        Object {
          "isUuid": "each value in include must be an UUID",
        }
      `);
      expect(result.value).toMatchInlineSnapshot(`
        Array [
          "non-uuid",
        ]
      `);
    });
  });
});

describe(`when changing by geo`, () => {
  describe(`when providing invalid shape`, () => {
    it(`should return error`, () => {
      expect(
        validateSync(getDtoWithInvalidMultiPolygon())[0].children,
      ).toMatchSnapshot();
    });
  });
});
