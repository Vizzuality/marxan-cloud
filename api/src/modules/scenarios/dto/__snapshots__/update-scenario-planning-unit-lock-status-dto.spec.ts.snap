// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when changing by geo when providing invalid shape should return error 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "IsFeatureCollectionOfPolygons": "Value must be a valid GeoJson of FeatureCollection of (Polygon/MultiPolygon)",
    },
    "property": "exclude",
    "target": PlanningUnitsByGeoJsonUpdateDto {
      "exclude": Array [
        Object {
          "features": Array [
            Object {
              "geometry": Object {
                "coordinates": Array [
                  Array [
                    Array [
                      Array [
                        102,
                        2,
                      ],
                      Array [
                        104,
                        5,
                      ],
                      Array [
                        103,
                        3,
                      ],
                    ],
                  ],
                ],
                "type": "MultiPolygon",
              },
              "properties": Object {},
              "type": "Feature",
            },
          ],
          "type": "FeatureCollection",
        },
      ],
      "include": Array [
        Object {
          "features": Array [
            Object {
              "geometry": Object {
                "coordinates": Array [
                  Array [
                    Array [
                      Array [
                        102,
                        2,
                      ],
                      Array [
                        103,
                        2,
                      ],
                      Array [
                        103,
                        3,
                      ],
                      Array [
                        102,
                        3,
                      ],
                      Array [
                        102,
                        2,
                      ],
                    ],
                  ],
                  Array [
                    Array [
                      Array [
                        100,
                        0,
                      ],
                      Array [
                        101,
                        0,
                      ],
                      Array [
                        101,
                        1,
                      ],
                      Array [
                        100,
                        1,
                      ],
                      Array [
                        100,
                        0,
                      ],
                    ],
                    Array [
                      Array [
                        100.2,
                        0.2,
                      ],
                      Array [
                        100.8,
                        0.2,
                      ],
                      Array [
                        100.8,
                        0.8,
                      ],
                      Array [
                        100.2,
                        0.8,
                      ],
                      Array [
                        100.2,
                        0.2,
                      ],
                    ],
                  ],
                ],
                "type": "MultiPolygon",
              },
              "properties": Object {},
              "type": "Feature",
            },
          ],
          "type": "FeatureCollection",
        },
      ],
    },
    "value": Array [
      Object {
        "features": Array [
          Object {
            "geometry": Object {
              "coordinates": Array [
                Array [
                  Array [
                    Array [
                      102,
                      2,
                    ],
                    Array [
                      104,
                      5,
                    ],
                    Array [
                      103,
                      3,
                    ],
                  ],
                ],
              ],
              "type": "MultiPolygon",
            },
            "properties": Object {},
            "type": "Feature",
          },
        ],
        "type": "FeatureCollection",
      },
    ],
  },
]
`;
